packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -293,26 +362,42 @@ function isFunctionParam(
   return false;
 }
 
-function isPromiseCatchCallWithHandler(
+function tryParseCatchCallWithhandler(
```

Comment:
[Naming] Also on the name, my instinct for things named _"try*"_ is that they literally use a `try`/`catch` and/or may throw an error. This doesn't. Perhaps a more direct name would be `parseCatchCallWithHandler`?

(same for the "then" equivalent later on in the file)